### PR TITLE
Add other random number generators

### PR DESCRIPTION
**Pull request by prbzrg**
I just added other random number generators

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -15,85 +14,99 @@ const rng = StableRNG(12345)
...
+            @test size(init(rng, 3, 4)) == (3, 4)
+            @test size(init(3, 4, 5)) == (3, 4, 5)
+            @test size(init(rng, 3, 4, 5)) == (3, 4, 5)
+            # Type
+            @test eltype(init(rng, 4, 2)) == Float32
+            @test eltype(init(4, 2)) == Float32
+            # RNG Closure
+            cl = init(rng)
+            @test cl(3) isa AbstractArray{Float32, 1}
+            @test cl(3, 5) isa AbstractArray{Float32, 2}
```
Comment (avik-pal) [inline test/runtests.jl]: Don't check `AbstractArray`. Have a list mapping the rngtype to the correct array type, and check against that. For ex, CURAND and CUDA should produce CuArray while the others need to produce Array

**Comment by prbzrg** [comment]
`zeros32` and `ones32` are ignoring `rng`. What should we do about them?
I think, we can have an extension for CUDA types. 🤔❓

**Comment by avik-pal** [comment]
Yeah fair point, we can do an extensions on them and create a CuArray if a CUDA RNG was passed

**Comment by prbzrg** [comment]
I made an extension for it, but `cl = init(rng)` dispatch incorrectly. It creates:
```julia
julia> CUDA.zeros(Float32)
0-dimensional CuArray{Float32, 0, CUDA.Mem.DeviceBuffer}:
0.0
```

**Comment by prbzrg** [comment]
How can we fix it?

File: Project.toml [hunk 2]
```diff
@@ -9,6 +9,12 @@ Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
 SpecialFunctions = "276daf66-3868-5448-9aa4-cd146d93841b"
 Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 
+[weakdeps]
+CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba"
+
+[extensions]
+CUDAExtWI = "CUDA"
```
Comment (avik-pal) [inline Project.toml]: Let's not use shortforms, it should be `WeightInitializersCUDAExt`

**Comment by avik-pal** [comment]
Changes pulled into https://github.com/LuxDL/WeightInitializers.jl/pull/9 and tested. The problem you had need a custom dispatch for the partial functions

**Pull request closed without merging**